python/cudf/cudf/core/column_accessor.py

```diff
@@ -243,25 +222,14 @@ def _grouped_data(self) -> abc.MutableMapping:
         else:
             return self._data
 
-    @cached_property
-    def _column_length(self):
-        try:
-            return len(self._data[next(iter(self._data))])
-        except StopIteration:
-            return 0
-
     def _clear_cache(self):
-        cached_properties = ("columns", "names", "_grouped_data")
+        cached_properties = ("columns", "names", "_grouped_data", "nrows")
         for attr in cached_properties:
             try:
                 self.__delattr__(attr)
             except AttributeError:
                 pass
 
-        # Column length should only be cleared if no data is present.
```

Comment:
It looks like we had an extra optimization in place where we only cleared the column length if there was no data present. I assume that would still be valid for nrows, but I don't recall all the places where `_clear_cache` is called off the top of my head. Could you double-check that? Most of the optimizations in `ColumnAccessor` are actually meaningful (which is not as true in the rest of the code base).

Reply:
Ah good catch. Yeah looks like this is called in methods where columns are modified, and these methods checks generally for row uniformity during modifications. I'll add this condition back for nrows